File: solutions/javascript/intersection-of-arrays.js
function intersect(a, b) {
  const freq = {};
  const result = [];
  for (let x of a) freq[x] = (freq[x] || 0) + 1;
  for (let x of b) {
    if (freq[x] > 0) {
      result.push(x);
      freq[x]--;
    }
  }
  return result;
}
console.log(intersect([1,2,2,1],[2,2]));

File: solutions/python/intersection-of-arrays.py
def intersect(a, b):
    freq = {}
    result = []
    for x in a:
        freq[x] = freq.get(x, 0) + 1
    for x in b:
        if x in freq and freq[x] > 0:
            result.append(x)
            freq[x] -= 1
    return result

print(intersect([1,2,2,1],[2,2]))